tools/const_finder/lib/const_finder.dart

```diff
@@ -72,12 +74,39 @@ class _ConstVisitor extends RecursiveVisitor<void> {
     super.visitConstructorInvocation(node);
   }
 
+  @override
+  void visitClass(Class node) {
+    // Check if this is a class that we should ignore.
+    if (!_isStaticIconProvider(node)) {
+      // Not an ignored class.
+      super.visitClass(node);
+      return;
+    }
+    inIgnoredClass = true;
+    super.visitClass(node);
+    inIgnoredClass = false;
+  }
+
+  // Constants from classes annotated with an instance of StaticIconProvider
+  // should be ignored.
+  bool _isStaticIconProvider(Class node) {
+    return node.annotations.any((Expression expression) {
+      if (expression is! ConstantExpression) {
+        return false;
+      }
+
+      final DartType type = expression.type;
+      return type is InterfaceType && type.classNode.name == 'StaticIconProvider';
```

Comment:
This makes sense. I wasn't able to hardcode the library URI and have the tests pass, since the test fixture doesn't import flutter/material (where I want to have the real annotation class), but passing it in via CLI option makes sense.

Reply:
done